src/adapters/index.ts

```diff
@@ -1,6 +1,18 @@
 import { ethers } from 'ethers'
 import { Chain } from '../core/constants/chains'
 import { IProtocolAdapter } from '../types/adapter'
+import { AaveV2ATokenPoolAdapter } from './aave-v2/products/pool/aaveV2ATokenPoolAdapter'
+import { AaveV2StableDebtTokenPoolAdapter } from './aave-v2/products/pool/aaveV2StableDebtTokenPoolAdapter'
+import { AaveV2StableVariableTokenPoolAdapter } from './aave-v2/products/pool/aaveV2StableVariableTokenPoolAdapter'
+import AAVE_V2_AVALANCHE_A_TOKEN_POOL_METADATA from './aave-v2/products/pool/avalanche/a-token-pool-metadata.json'
+import AAVE_V2_AVALANCHE_STABLE_DEBT_TOKEN_POOL_METADATA from './aave-v2/products/pool/avalanche/stable-debt-token-pool-metadata.json'
+import AAVE_V2_AVALANCHE_VARIABLE_DEBT_TOKEN_POOL_METADATA from './aave-v2/products/pool/avalanche/variable-debt-token-pool-metadata.json'
+import AAVE_V2_ETHEREUM_A_TOKEN_POOL_METADATA from './aave-v2/products/pool/ethereum/a-token-pool-metadata.json'
+import AAVE_V2_ETHEREUM_STABLE_DEBT_TOKEN_POOL_METADATA from './aave-v2/products/pool/ethereum/stable-debt-token-pool-metadata.json'
+import AAVE_V2_ETHEREUM_VARIABLE_DEBT_TOKEN_POOL_METADATA from './aave-v2/products/pool/ethereum/variable-debt-token-pool-metadata.json'
+import AAVE_V2_POLYGON_A_TOKEN_POOL_METADATA from './aave-v2/products/pool/polygon/a-token-pool-metadata.json'
+import AAVE_V2_POLYGON_STABLE_DEBT_TOKEN_POOL_METADATA from './aave-v2/products/pool/polygon/stable-debt-token-pool-metadata.json'
+import AAVE_V2_POLYGON_VARIABLE_DEBT_TOKEN_POOL_METADATA from './aave-v2/products/pool/polygon/variable-debt-token-pool-metadata.json'
```

Comment:
This is getting out of hand and it'll get worse with more protocols and chains. Not urgent, but it'll be much nicer to have, per adapter, a `getMetadata` function or a class. That could code be autogenerated, since we have access to all the files when we build metadata and could create a method/entry for each with recast.

It would also simplify the `supportedProtocols` object, because once all adapters have the same input:
```
{
  metadataFetcher: MetadataFetcher
  chainId: Chain
  provider: ethers.providers.StaticJsonRpcProvider
}
```

We will be able to stop repeating the code and simplify it such as:
```
{
  [Protocol.Stargate]: {
    [Chain.Ethereum]: [
      StargatePoolAdapter,
      StargateVestingAdapter,
    ],
    [Chain.Arbitrum]: [
      StargatePoolAdapter,
    ],
  },

  [Protocol.Example]: {
    [Chain.Ethereum]: [
      ExampleProductAdapter,
    ],
  },

  [Protocol.AaveV2]: {
    [Chain.Ethereum]: [
      AaveV2ATokenPoolAdapter,
      AaveV2StableDebtTokenPoolAdapter,
      AaveV2StableVariableTokenPoolAdapter,
    ],
    [Chain.Polygon]: [
      AaveV2ATokenPoolAdapter,
      AaveV2StableDebtTokenPoolAdapter,
      AaveV2StableVariableTokenPoolAdapter,
    ],
    [Chain.Avalanche]: [
      AaveV2ATokenPoolAdapter,
      AaveV2StableDebtTokenPoolAdapter,
      AaveV2StableVariableTokenPoolAdapter,
    ],
  },
}
```